pkg/nvidianetwork/nicclusterpolicy.go

```diff
@@ -0,0 +1,258 @@
+
+		builder.errorMsg = "ClusterPolicy 'Object.Definition' is nil"
+	}
+
+	return &builder
+}
+
+// Get returns nicclusterPolicy object if found.
+func (builder *Builder) Get() (*nvidianetworkv1alpha1.NicClusterPolicy, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof(
+		"Collecting NicClusterPolicy object %s", builder.Definition.Name)
+
+	nicClusterPolicy := &nvidianetworkv1alpha1.NicClusterPolicy{}
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name: builder.Definition.Name,
+	}, nicClusterPolicy)
+
+	if err != nil {
+		glog.V(100).Infof(
+			"NicClusterPolicy object %s doesn't exist", builder.Definition.Name)
+
+		return nil, err
+	}
+
+	return nicClusterPolicy, err
+}
+
+// Pull loads an existing NicClusterPolicy into Builder struct.
+func Pull(apiClient *clients.Settings, name string) (*Builder, error) {
+	glog.V(100).Infof("Pulling existing nicClusterPolicy name: %s", name)
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &nvidianetworkv1alpha1.NicClusterPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("NicClusterPolicy name is empty")
+
+		builder.errorMsg = "NicClusterPolicy 'name' cannot be empty"
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("NicClusterPolicy object %s doesn't exist", name)
+	}
+
+	builder.Definition = builder.Object
+
+	return &builder, nil
+}
+
+// Exists checks whether the given NicClusterPolicy exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof(
+		"Checking if NicClusterPolicy %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	if err != nil {
+		glog.V(100).Infof("Failed to collect NicClusterPolicy object due to %s", err.Error())
+	}
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Delete removes a NicClusterPolicy.
+func (builder *Builder) Delete() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting NicClusterPolicy %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("nicclusterpolicy cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Delete(context.TODO(), builder.Definition)
+
+	if err != nil {
+		return builder, fmt.Errorf("cannot delete nicclusterpolicy: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// Create makes a NicClusterPolicy in the cluster and stores the created object in struct.
+func (builder *Builder) Create() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the NicClusterPolicy %s", builder.Definition.Name)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+
```

Comment:
maybe we should add a log here? might be good for debugging in the future

Reply:
Will add. Thx.